python/ray/tests/test_node_death.py

```diff
@@ -25,5 +26,45 @@ def test_normal_termination(ray_start_cluster):
     assert worker_node_info["DeathReasonMessage"] == "Received SIGTERM"
 
 
+def test_abnormal_termination(ray_start_cluster):
+    cluster = ray_start_cluster
+    cluster.add_node(resources={"head": 1})
+    ray.init(address=cluster.address)
+    worker_node = cluster.add_node(resources={"worker": 1})
+    cluster.wait_for_nodes()
+
+    @ray.remote
+    def get_node_id():
+        return ray.get_runtime_context().get_node_id()
+
+    head_node_id = ray.get(get_node_id.options(resources={"head": 1}).remote())
+    worker_node_id = ray.get(get_node_id.options(resources={"worker": 1}).remote())
+
+    wait_for_condition(
+        lambda: {node["NodeID"] for node in ray.nodes() if (node["Alive"])}
+        == {head_node_id, worker_node_id}
+    )
+
+    # Simulate the worker node crashes.
+    cluster.remove_node(worker_node, False)
+
+    # Use a larger timeout to wait for GCS health checker
+    # marks the worker node as dead.
```

Comment:
To make the test run faster, we can change the health check config in the test so that GCS discover node death faster

Reply:
Added fast node detection.